### DISCUSSION TITLE
Abandoned checkout automation does not trigger with Storefront API

### DISCUSSION THREAD

**Discussion by martynasj** [Shopify/storefront-api-feedback#246] [Help] [answered]
We have a headless store which uses the cart API to redirect user to the shopify hosted checkout page. When checkout is abandoned, we can see it in the shopify admin panel next to "Abandoned checkouts". However, the "abandoned checkout" automation trigger does not work. 

Is there any specific reason why it doesn't work with headless, but it does work with the "buy button"?

Thanks for the update!

**Comment by vixdug** [comment, marked as answer]
Hey @martynasj You can setup your marketing automation to run for Headless. I'm using App Title here as my condition, but you can choose whatever works best for your business: 
![image](https://github.com/user-attachments/assets/a07bb960-e8d7-4aeb-ad9c-a9ab379b29b4)

Make sure you are opted in to the newest version: https://help.shopify.com/en/manual/promoting-marketing/create-marketing/migrate-abandoned-checkout
